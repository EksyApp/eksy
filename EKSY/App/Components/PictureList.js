import React, {Component} from 'react'
import {ListView} from 'react-native'
import PropTypes from 'prop-types'
import Picture from './Picture'

class PictureList extends Component {
	
	constructor(props) {
		super(props)
		
		const datasource = new ListView.DataSource({rowHasChanged: (r1,r2) => r1.uri !== r2.uri})
		this.state = {
			dataSource: datasource.cloneWithRows(this.props.data)
		}
		
	}
	
	render() {
		console.log(this.props)
		return (
				<ListView
						dataSource={this.state.dataSource}
				    renderRow = {(picture) => <Picture uri={picture.uri} />}
				/>
		)
	}
}

PictureList.propTypes = {
	data: PropTypes.arrayOf(PropTypes.shape({
		uri: PropTypes.string.isRequired
	})),
}

export default PictureList