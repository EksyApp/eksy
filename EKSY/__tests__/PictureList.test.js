import 'react-native';
import React from 'react';
import PictureList from '../App/Components/PictureList';
import renderer from 'react-test-renderer';

test('picturelist renders correctly', () => {
  const tree = renderer.create(
    <PictureList data = {[{width: 320, height: 240, uri: 'http://www.catpic.com'}]}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
