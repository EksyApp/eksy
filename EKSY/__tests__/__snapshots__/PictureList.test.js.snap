// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`picturelist renders correctly 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 1,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={undefined}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          undefined,
          Array [
            Object {
              "marginTop": 10,
            },
            undefined,
          ],
          Object {
            "height": 240,
            "width": 320,
          },
        ]
      }
    >
      <Image
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        source={
          Object {
            "uri": "http://www.catpic.com",
          }
        }
        style={
          Array [
            Object {
              "resizeMode": "contain",
            },
            Object {
              "height": 240,
              "width": 320,
            },
          ]
        }
        testID={undefined}
      />
    </View>
  </View>
</RCTScrollView>
`;
